test/e2e/network/kube_proxy.go

```diff
@@ -182,6 +182,11 @@ var _ = SIGDescribe("KubeProxy", func() {
 			kubeProxyE2eImage))
 		fr.PodClient().CreateSync(serverPodSpec)
 
+		// The host exec pod should be ready before we start the test, as it is time sensitive
+		// (if the host exec pod takes a long time to be ready, that could cause spurious timeouts)
+		if readyErr := e2epod.WaitForPodsReady(fr.ClientSet, fr.Namespace.Name, hostExecPod.Name, 0); readyErr != nil {
```

Comment:
:thinking: I thought that the `CreateSync()` with the `WaitForPodNameRunningInNamespace` was enough
https://github.com/kubernetes/kubernetes/blob/5a382b2d836fb7f57fd1631251651d00c802645b/test/e2e/framework/pods.go#L99-L108

Reply:
if that is not the case then I think we should modify the CreateSync os everybody using it got the right behaviour

Reply:
what do you thing about replacing the `e2epod.WaitForPodNameRunningInNamespace`  in `CreateSync` by 
```
// WaitTimeoutForPodReadyInNamespace waits the given timeout diration for the
// specified pod to be ready and running.
func WaitTimeoutForPodReadyInNamespace(c clientset.Interface, podName, namespace string, timeout time.Duration) error {
	return wait.PollImmediate(poll, timeout, podRunningAndReady(c, podName, namespace))
}
```
or better do this in another PR?

Reply:
scratch this, it already has a WaitForReady in another places ...

Reply:
https://github.com/kubernetes/kubernetes/pull/99147